test/multiplier/namespace.go

```diff
@@ -21,9 +21,26 @@ func (n *namespace) Name() Name {
 }
 
 func (n *namespace) Apply(source action.Actions) action.Actions {
-	result := make(action.Actions, 1, len(source)+1)
-	result[0] = action.Namespace([]byte("/example"))
-	result = append(result, source...)
+	lastStoreWrappingIndex := 0
+	result := make(action.Actions, len(source)+1)
+
+	for i, a := range source {
+		switch a.(type) {
+		case *action.NewStore, *action.NewBadgerStore, *action.NewMemoryStore:
+			lastStoreWrappingIndex = i
+		}
+	}
+
+	offset := 0
+	for i, a := range source {
+		newIndex := i + offset
+		result[newIndex] = a
+
+		if i == lastStoreWrappingIndex {
+			result[newIndex+1] = action.Namespace([]byte("/example"))
+			offset += 1
+		}
+	}
```

Comment:
suggestion: You could avoid doing this loop with 
```go
source := append(source, nil)
copy(source[lastStoreWrappingIndex + 1:], source[lastStoreWrappingIndex:])
source[lastStoreWrappingIndex] = action.Namespace([]byte("/example"))
```

Reply:
The suggestion involves fewer lines, but I'm not sure the twin re-slicing and a copy method it is easier to read.

Leaving as is for now.